Source/WebCore/Configurations/WebCore.xcconfig

```diff
@@ -51,11 +51,12 @@ CLANG_INSTRUMENT_FOR_OPTIMIZATION_PROFILING_ENABLE_LLVM_PROFILE_GENERATION = YES
 FRAMEWORK_SEARCH_PATHS[sdk=embedded*] = $(WK_QUOTED_OVERRIDE_FRAMEWORKS_DIR) $(BUILT_PRODUCTS_DIR);
 FRAMEWORK_SEARCH_PATHS[sdk=macosx*] = $(WK_QUOTED_OVERRIDE_FRAMEWORKS_DIR);
 
-SYSTEM_FRAMEWORK_SEARCH_PATHS = $(inherited) $(SDK_DIR)$(SYSTEM_LIBRARY_DIR)/PrivateFrameworks $(SDK_DIR)$(SYSTEM_LIBRARY_DIR)/Frameworks
+SYSTEM_FRAMEWORK_SEARCH_PATHS = $(inherited) $(WK_PRIVATE_SDK_DIR)$(SYSTEM_LIBRARY_DIR)/PrivateFrameworks $(SDK_DIR)$(SYSTEM_LIBRARY_DIR)/Frameworks $(SYSTEM_FRAMEWORK_SEARCH_PATHS_APPLEJPEGXL_$(WK_USE_APPLEJPEGXL));
+SYSTEM_FRAMEWORK_SEARCH_PATHS_APPLEJPEGXL_YES = $(SDK_DIR_WebKitSDKAdditions)$(SYSTEM_LIBRARY_DIR)/PrivateFrameworks;
```

Comment:
This reads strangely - we use WebKitSDKAdditions whenever we use AppleJPEGXL? But we shouldn't be using SDK additions for it on macOS Sonoma. Perhaps something is misnamed here. Or are you adding it unconditionally, and relying on search order to pick up the base SDK copy? That would seem fragile.

Reply:
> we use WebKitSDKAdditions whenever we use AppleJPEGXL?

The converse: Whenever we use AppleJPEGXL, we use WebKitSDKAdditions. This is because AppleJPEGXL.framework is downlevel only.

> But we shouldn't be using SDK additions for it on macOS Sonoma.

Further up in this file, we have
```
WK_USE_APPLEJPEGXL = $(WK_USE_APPLEJPEGXL_$(WK_PLATFORM_NAME));
WK_USE_APPLEJPEGXL_macosx = $(WK_USE_APPLEJPEGXL$(WK_MACOS_1400));
WK_USE_APPLEJPEGXL_MACOS_BEFORE_1400 = YES;
```

Anyways, I think that what I'd like to do is remove this part of the patch, and keep the search paths in SDKSettings.plist for internal SDK additions only. This resticts the scope of this PR to only changing how the _public_ SDK additions hook into search paths.